### PR TITLE
 Update scripts to generate AGPL headers

### PR DESCRIPTION
**Pull request by vanwagonet**

### CODE REVIEW

**Comment by vanwagonet** [comment]
You may want to review the first commit separately. 😬

File: scripts/update-headers.js [hunk 1]
```diff
@@ -96,40 +86,35 @@ function check(files, skipWrite = true) {
...
+      const updated = `${shebang}${newBanner(file, banner)}${noBang}`
       if (!skipWrite) {
-        writeFileSync(file, text, 'utf8')
+        writeFileSync(file, updated, 'utf8')
       }
       replaced.push(file)
       continue
     }
 
-    if (!hasBanner) {
```
Comment (vanwagonet) [inline scripts/update-headers.js]: if there was no banner, then the above no copyright case would have already been run. This was unreachable.

File: scripts/update-headers.js [hunk 2]
```diff
@@ -96,40 +86,35 @@ function check(files, skipWrite = true) {
...
+      if (banner.includes('GPL')) {
         incompatible.push(file)
       } else {
         skipped.push(file)
       }
       continue
     }
 
-    const updated = text.replace(bannerExp, banner)
+    const updated = text.replace(headerExp, `$1${newBanner(file, banner)}`)
```
Comment (vanwagonet) [inline scripts/update-headers.js]: `$1` is the shebang. We want to replace everything else in the header with the new banner.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 95.34% | 95.34% | 0%
**Student** | 90.78% | 90.78% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.78% | 92.78% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: scripts/update-headers.js [hunk 3]
```diff
@@ -19,73 +37,45 @@ program
...
-// Doing this to be safe, so we don't replace other comments
-const copyrightExp = /(^|\r\n|\n|\r)[\/*#|\s]*Copyright\s+[^\s\r\n]+/i
-
-// This regex is used to find and/or replace the first block comment in a file
-const bannerExp = /(^|\r\n|\n|\r)(?:\/\/[^\n\r]*(?:\r\n|\n|\r))*\/\/[^\n\r]*($|\r\n|\n|\r)/i
-
-// Used to test whether the Instructure appears in a file
-const instructureExp = /(^|\r\n|\n|\r)*Instructure/i
+//                   optional shebang   space    star block        hash or slash block       space
+const headerExp = /^(#!.*(?:\r?\n|\r))?[\s\r\n]*(\/\*.*?\*\/|(?:(?:#|\/\/).*(?:\r?\n|\r))+)?[\s\r\n]*/
```
Comment (bluwave) [inline scripts/update-headers.js]: holy moly regex batman! 😱

**Comment by bluwave** [comment]
i really just looked at your first commit and as many headers as i could :)

**Pull request merged**